feature-wallet-impl/src/main/java/io/novafoundation/nova/feature_wallet_impl/data/network/blockchain/assets/transfers/utility/NativeAssetTransfers.kt

```diff
@@ -11,20 +15,64 @@ import io.novafoundation.nova.feature_wallet_api.domain.validation.EnoughTotalTo
 import io.novafoundation.nova.feature_wallet_api.domain.validation.PhishingValidationFactory
 import io.novafoundation.nova.feature_wallet_impl.data.network.blockchain.assets.transfers.BaseAssetTransfers
 import io.novafoundation.nova.runtime.ext.accountIdOrDefault
+import io.novafoundation.nova.runtime.ext.isCommissionAsset
 import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
 import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.storage.source.StorageDataSource
+import jp.co.soramitsu.fearless_utils.hash.isPositive
+import jp.co.soramitsu.fearless_utils.runtime.RuntimeSnapshot
 import jp.co.soramitsu.fearless_utils.runtime.extrinsic.ExtrinsicBuilder
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module
+import jp.co.soramitsu.fearless_utils.runtime.metadata.storage
+import jp.co.soramitsu.fearless_utils.runtime.metadata.storageKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
 
 class NativeAssetTransfers(
     chainRegistry: ChainRegistry,
     assetSourceRegistry: AssetSourceRegistry,
     extrinsicService: ExtrinsicService,
     phishingValidationFactory: PhishingValidationFactory,
-    enoughTotalToStayAboveEDValidationFactory: EnoughTotalToStayAboveEDValidationFactory
+    enoughTotalToStayAboveEDValidationFactory: EnoughTotalToStayAboveEDValidationFactory,
+    private val storageDataSource: StorageDataSource,
+    private val accountRepository: AccountRepository,
 ) : BaseAssetTransfers(chainRegistry, assetSourceRegistry, extrinsicService, phishingValidationFactory, enoughTotalToStayAboveEDValidationFactory) {
 
     override val validationSystem: AssetTransfersValidationSystem = defaultValidationSystem()
 
+    override suspend fun totalCanDropBelowMinimumBalance(chainAsset: Chain.Asset): Boolean {
+        if (!chainAsset.isCommissionAsset) return false
+
+        val chain = chainRegistry.getChain(chainAsset.chainId)
+        val metaAccount = accountRepository.getSelectedMetaAccount()
+
+        val accountInfo = storageDataSource.query(
+            chainAsset.chainId,
+            keyBuilder = { getAccountInfoStorageKey(metaAccount, chain, it) },
+            binding = { it, runtime -> it?.let { bindAccountInfo(it, runtime) } }
+        )
+
+        return accountInfo != null && accountInfo.consumers.isPositive()
```

Comment:
Afair we can only reap account (drop balance below minimum) if consumers is zero?

Reply:
Yes, fixed